src/resolve/resolved-varlink.c

```diff
@@ -157,45 +173,23 @@ static bool validate_and_mangle_flags(
         return true;
 }
 
-static void vl_method_resolve_hostname_complete(DnsQuery *query) {
-        _cleanup_(dns_resource_record_unrefp) DnsResourceRecord *canonical = NULL;
-        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
-        _cleanup_(dns_query_freep) DnsQuery *q = query;
-        _cleanup_free_ char *normalized = NULL;
+static int find_addr_records(
+                JsonVariant **array,
+                DnsQuestion *question,
+                DnsQuery *q,
+                DnsResourceRecord **canonical,
+                bool search_domain_flag) {
```

Comment:
hmmm, why not make this parameter a `const char* search_domain`? i.e. already call DNS_SEARCH_DOMAIN_NAME() in the caller where appropriate and pass NULL here if not appropriate. I think that would be more readable.